domain.go

```diff
@@ -1832,10 +1832,11 @@ func (d *Domain) ListAllInterfaceAddresses(src uint) ([]DomainInterface, error)
 
 		for k := 0; k < numAddr; k++ {
 			ifaces[i].Addrs[k] = DomainIPAddress{}
-			ifaces[i].Addrs[k].Type = int(addrSlice[k]._type)
-			ifaces[i].Addrs[k].Addr = C.GoString(addrSlice[k].addr)
-			ifaces[i].Addrs[k].Prefix = uint(addrSlice[k].prefix)
-
+			if addSlice[k] != nil {
```

Comment:
I think that's a symptom of an earlier problem. Can you try the following diff instead and tell me if it works:

@@ -1821,14 +1821,9 @@ func (d *Domain) ListAllInterfaceAddresses(src uint) ([]DomainInterface, error)
                ifaces[i].Hwaddr = C.GoString(ifaceSlice[i].hwaddr)
 
                numAddr := int(ifaceSlice[i].naddrs)
-               addrHdr := reflect.SliceHeader{
-                       Data: uintptr(unsafe.Pointer(&ifaceSlice[i].addrs)),
-                       Len:  int(numAddr),
-                       Cap:  int(numAddr),
-               }
 
                ifaces[i].Addrs = make([]DomainIPAddress, numAddr)
-               addrSlice := *(*[]C.virDomainIPAddressPtr)(unsafe.Pointer(&addrHdr))
+               addrSlice := *(*[]C.virDomainIPAddressPtr)(unsafe.Pointer(&ifaceSlice[i].addrs))
 
                for k := 0; k < numAddr; k++ {
                        ifaces[i].Addrs[k] = DomainIPAddress{}